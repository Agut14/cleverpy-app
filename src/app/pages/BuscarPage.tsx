import { useForm } from '../../hooks/useForm';
import '../styles/buscarPage.scss'
import { useLocation, useNavigate } from 'react-router-dom';
import queryString from 'query-string';
import { useSearch } from '../hooks/useSearch';
import { RiDoubleQuotesL, RiDoubleQuotesR } from "react-icons/ri";


export const BuscarPage = () => {

  const { searchText, onHandleChange, resetData } = useForm({
    searchText: ''
  });
  const { getPostsByAuthor } = useSearch();
  const navigate = useNavigate();
  const location = useLocation();

  const query = queryString.parse( location.search );
  const searchValue = query.q ?? '';
  const posts = getPostsByAuthor( String(searchValue));

  const showSearch = (searchValue.length === 0);
  const showError = (searchValue.length !== 0) && posts?.length === 0;
  
  

  const onSubmitSearch = ( event: React.FormEvent ) => {
    event.preventDefault();
    if(searchText?.trim() == '') return;
    navigate(`?q=${searchText?.toLowerCase().trim()}`);
    resetData();
  };

  return (
    <div className="buscar-page-content">
      <div className='form-search-posts'>
        <div>
          <h2>Buscar posts por usuario</h2>
          <form onSubmit={(event) => onSubmitSearch(event) }>
            <input
              className='input-global' 
              type="text"
              placeholder="Escribe algo..."
              name="searchText"
              autoComplete="off"
              value={ searchText }
              onChange={ (event) => onHandleChange(event.target)}
              />
            <button className="btn-buscar">
              Buscar
            </button>  
          </form>
        </div>
      </div>

      <div>
      <h2>Resultados</h2>
      
      { showSearch &&
      <div>
        <h4>Prueba a buscar por el nombre del usuario, por ejemplo Graham...</h4>
      </div> 
      }
      
      { showError &&
      <div>
        <h4>No hay resultados para <b>{ searchValue }</b></h4>
      </div> 
      }

      <div className='post-search'>
          <div>{ posts?.map(post => (
            <div key={post.id} className='post-search-item'><RiDoubleQuotesL /> { post.body } <RiDoubleQuotesR /></div>
          )) }</div>
        </div>
      </div>
    </div>




  )
}
