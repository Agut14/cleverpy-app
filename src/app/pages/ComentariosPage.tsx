

export const ComentariosPage = () => {
  return (
    <div>ComentariosPage</div>
  )
}
