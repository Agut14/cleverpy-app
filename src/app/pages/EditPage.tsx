import { useParams } from "react-router-dom"
import { usePosts } from '../hooks/usePosts';
import '../styles/editPageStyles.scss'
import { useForm } from '../../hooks/useForm';
import { post } from "../../interfaces/postInterface";





export const EditPage = () => {

    const { id } = useParams();

   const { posts, getPostById, deletePost, updatePostsFromHook } = usePosts();
  //  const post = getPostById(id); es una forma de hacerlo
  //aquÃ­ va otr
  
  const post = posts.find(post => String(post.id) === id);
  
   const { body, title, onHandleChange } = useForm({
    body: post?.body,
    title: post?.title
   });


   const formData: post = {
    userId: post?.userId,
    id: post?.id,
    title: title,
    body: body
   }

   const isDisabled = false;
    
    
    return (
      <div className="edit-page-content">
          <div className="edit-form">
            <h3>"{ post?.title }"</h3>
            

            <form action="#">
              <fieldset className="edit-form-fields" disabled={isDisabled}>
                <legend>Editar post con id { post?.id }</legend>
                <label className="edit-page-label" htmlFor="body">Title</label>
                <textarea 
                  name="title"
                  value={ title } 
                  onChange={(event) => onHandleChange(event.target) }/>
                <label className="edit-page-label" htmlFor="body">Body</label>
                <textarea 
                  name="body"
                  value={ body }
                  onChange={(event) => onHandleChange(event.target) } />
              </fieldset>
              <div className="edit-page-buttons">
                  { isDisabled ? <button>Editar</button> : <button onClick={ (event) => updatePostsFromHook(event, post?.id, JSON.stringify(formData)) }>Guardar</button>}
                  <button onClick={ (event)=> deletePost(event)}>Borrar</button>
              </div>

              
            </form>
          </div>
      </div>
)
 
}
