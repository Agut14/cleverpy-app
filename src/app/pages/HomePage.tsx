import { useAppDispatch } from '../../hooks/index';
import { useSelector } from 'react-redux';
import { RootState } from '../../store/store';
import { useEffect } from 'react';
import { getPosts } from '../slices/post/thunks';
import '../styles/cardStyles.scss'


export const HomePage = () => {

  const dispatch = useAppDispatch();
  const { posts, isLoading } = useSelector( (state: RootState ) => state.posts);

  useEffect(() => {
    dispatch(getPosts());
  }, []);


  return (
    <>
    <div className='home-content'>
      {posts.map( post => (
        <div className='card'>
          <div className='card-title'>
            <h3>{ post.title }</h3>
          </div>
          <div className='card-body'>
             <p>{ post.body }</p>
          </div>
        </div>
      ))}
      
    </div>
    </>
    
  )
}
