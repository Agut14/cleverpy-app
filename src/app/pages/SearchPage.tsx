import { useForm } from '../../hooks/useForm';
import '../styles/buscarPage.scss'
import { NavLink, useLocation, useNavigate } from 'react-router-dom';
import queryString from 'query-string';
import { useSearch } from '../hooks/useSearch';
import { RiDoubleQuotesL, RiDoubleQuotesR } from "react-icons/ri";
import { FiEdit3 } from "react-icons/fi";


export const SearchPage = () => {

  const { searchText, onHandleChange, resetData } = useForm({
    searchText: ''
  });
  const { getPostsByAuthor } = useSearch();
  const navigate = useNavigate();
  const location = useLocation();

//se obtiene el valor de búsquede de la url
  const query = queryString.parse( location.search );
  const searchValue = query.q ?? '';
//se obtienen los posts
  const posts = getPostsByAuthor( String(searchValue));

  const showSearch = (searchValue.length === 0);
  const showError = (searchValue.length !== 0) && posts?.length === 0;
  
  
//cuando se realiza la búsqueda, se llama a la url actual con el
//param q=searchValue
  const onSubmitSearch = ( event: React.FormEvent ) => {
    event.preventDefault();
    if(searchText?.trim() === '') return;
    navigate(`?q=${searchText?.toLowerCase().trim()}`);
    resetData();
  };

  return (
    <div className="buscar-page-content">
      <div className='form-search-posts'>
        <div>
          <h2 className='h2-search'>Buscar posts por usuario</h2>
          <form onSubmit={(event) => onSubmitSearch(event) }>
            <input
              className='input-global' 
              type="text"
              placeholder="Escribe algo..."
              name="searchText"
              autoComplete="off"
              value={ searchText }
              onChange={ (event) => onHandleChange(event.target)}
              />
            <button className="btn-buscar">
              Buscar
            </button>  
          </form>
        </div>
      </div>

      <div className='search-results'>
      <h2 className='h2-search'>Resultados</h2>
      
      { showSearch &&
      <div>
        <h4 className='h4-search'>Prueba a buscar por el nombre del usuario, por ejemplo Graham...</h4>
      </div> 
      }
      
      { showError &&
      <div>
        <h4 className='h4-search'>No hay resultados para <b>{ searchValue }</b></h4>
      </div> 
      }

      <div className='post-search'>
          <div>{ posts?.map(post => (
            <div aria-label='search-container' key={post.id} className='post-search-item'>
              <RiDoubleQuotesL /> { post.body } <RiDoubleQuotesR />
              <NavLink to={`/edit/${post.id}`}>
                <button className='edit-search-page'><FiEdit3 /></button>
              </NavLink>
            </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  )
}
