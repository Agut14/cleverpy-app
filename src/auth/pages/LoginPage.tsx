import '../styles/loginStyles.scss';
import { FaUserAlt } from "react-icons/fa";
import { RiLockPasswordFill } from "react-icons/ri";
import { useNavigate } from 'react-router-dom';
import { useContext } from 'react';
import { AuthContext } from '../../context/AuthContext';
import { useForm } from '../../hooks/useForm';


export const LoginPage = () => {

  const { authLogin } = useContext(AuthContext)
  const navigate = useNavigate();
  const { username, sendData, onHandleChange } = useForm({
    username: ''
  });

  const onLoginUser = (event: React.FormEvent, username: string = '') => {
    event.preventDefault();
    if( username.trim().length > 0) {
      authLogin( username );
    navigate('/', {
      replace: true
    });
    }
  }

  return (
    <>
    <div className='login-container'>
      <div className='login-subcontainer'>
      <form className='login-form' onSubmit={ (event) => onLoginUser(event, username) }>
        <div className="flex-row">
          <label className="lf--label" htmlFor="username">
            <FaUserAlt />
          </label>
          <input 
            id="username" 
            className='lf--input' 
            placeholder='Usuario'
            name="username" 
            type='text'
            value={ username }
            onChange={(event) => onHandleChange(event.target)}/>
        </div>
        <input 
          className='lf--submit' 
          type='submit' 
          value='LOGIN'/>
      </form>
      </div>
    </div>
    </>
    
  )
}
