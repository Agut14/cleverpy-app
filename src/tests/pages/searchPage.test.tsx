
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../app/pages/SearchPage';
import { Provider } from 'react-redux';
import { store } from '../../store/store';


export{};

describe('Buscar page testings', () => {

    
    test('renders default values correctly', () => {

        const { container } = render(
            <Provider store={store}>
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
            </Provider>
        )
        
        expect( container ).toMatchSnapshot();

    })
})