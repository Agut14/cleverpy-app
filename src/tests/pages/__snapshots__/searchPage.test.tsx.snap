// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buscar page testings renders default values correctly 1`] = `
<div>
  <div
    class="buscar-page-content"
  >
    <div
      class="form-search-posts"
    >
      <div>
        <h2
          class="h2-search"
        >
          Buscar posts por usuario
        </h2>
        <form>
          <input
            autocomplete="off"
            class="input-global"
            name="searchText"
            placeholder="Escribe algo..."
            type="text"
            value=""
          />
          <button
            class="btn-buscar"
          >
            Buscar
          </button>
        </form>
      </div>
    </div>
    <div
      class="search-results"
    >
      <h2
        class="h2-search"
      >
        Resultados
      </h2>
      <div>
        <h4
          class="h4-search"
        >
          Prueba a buscar por el nombre del usuario, por ejemplo Graham...
        </h4>
      </div>
      <div
        class="post-search"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
